### ISSUE TITLE
Kurento 6.12.0 crash on ubuntu16.04

### ISSUE THREAD

**Issue by syamgk** [Kurento/bugtracker#429]
Error log:
```
Stack trace:
[export_srtp_keys]
/workdir/ext/dtls/gstdtlsconnection.c:631
[gst_dtls_connection_process]
/workdir/ext/dtls/gstdtlsconnection.c:512
[process_buffer]
/workdir/ext/dtls/gstdtlsdec.c:498
[sink_chain]
/workdir/ext/dtls/gstdtlsdec.c:579
[gst_pad_chain_data_unchecked]
/workdir/gst/gstpad.c:4185
[gst_pad_push]
/workdir/gst/gstpad.c:4556
[sink_chain]
/workdir/ext/dtls/gstdtlssrtpdemux.c:128
[gst_pad_chain_data_unchecked]
/workdir/gst/gstpad.c:4185
[gst_pad_push]
/workdir/gst/gstpad.c:4556
[gst_proxy_pad_chain_default]
/workdir/gst/gstghostpad.c:127
[gst_pad_chain_data_unchecked]
/workdir/gst/gstpad.c:4185
[gst_pad_push]
/workdir/gst/gstpad.c:4556
[gst_base_src_loop]
/workdir/libs/gst/base/gstbasesrc.c:2851
[gst_task_func]
/workdir/gst/gsttask.c:346
[g_thread_pool_thread_proxy]
/build/glib2.0-pjKWYQ/glib2.0-2.48.2/./glib/gthreadpool.c:308
[g_thread_proxy]
/build/glib2.0-pjKWYQ/glib2.0-2.48.2/./glib/gthread.c:780
```
**KMS Version**:

`
Kurento Media Server version: 6.12.0
Found modules:
        'core' version 6.12.0
        'elements' version 6.12.0
        'filters' version 6.12.0
`

**Ubuntu Version**
Operating system:
`
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial
`

**Other libraries versions**:

```
dpkg -l | grep -Pi 'kurento|gst.*1.5|nice' 
ii  gstreamer1.5-alsa:amd64               1.8.1-1kurento1.16.04                           amd64        GStreamer plugin for ALSA
ii  gstreamer1.5-libav:amd64              1.8.1-1kurento1.16.04                           amd64        libav plugin for GStreamer
ii  gstreamer1.5-libav-dbg:amd64          1.8.1-1kurento1.16.04                           amd64        libav plugin for GStreamer (debug symbols)
ii  gstreamer1.5-nice:amd64               0.1.15-1kurento3.16.04                          amd64        ICE library (GStreamer 1.5 plugin)
ii  gstreamer1.5-nice-dbgsym:amd64        0.1.15-1kurento3.16.04                          amd64        debug symbols for package gstreamer1.5-nice
ii  gstreamer1.5-plugins-bad:amd64        1.8.1-1kurento4.16.04                           amd64        GStreamer plugins from the "bad" set
ii  gstreamer1.5-plugins-bad-dbg:amd64    1.8.1-1kurento4.16.04                           amd64        GStreamer plugins from the "bad" set (debug symbols)
ii  gstreamer1.5-plugins-base:amd64       1.8.1-1kurento1.16.04                           amd64        GStreamer plugins from the "base" set
ii  gstreamer1.5-plugins-base-dbg:amd64   1.8.1-1kurento1.16.04                           amd64        GStreamer plugins from the "base" set
ii  gstreamer1.5-plugins-good:amd64       1.8.1-1kurento2.16.04                           amd64        GStreamer plugins from the "good" set
ii  gstreamer1.5-plugins-good-dbg:amd64   1.8.1-1kurento2.16.04                           amd64        GStreamer plugins from the "good" set
ii  gstreamer1.5-plugins-ugly:amd64       1.8.1-1kurento1.16.04                           amd64        GStreamer plugins from the "ugly" set
ii  gstreamer1.5-plugins-ugly-dbg:amd64   1.8.1-1kurento1.16.04                           amd64        GStreamer plugins from the "ugly" set (debug symbols)
ii  gstreamer1.5-pulseaudio:amd64         1.8.1-1kurento2.16.04                           amd64        GStreamer plugin for PulseAudio
ii  gstreamer1.5-x:amd64                  1.8.1-1kurento1.16.04                           amd64        GStreamer plugins for X11 and Pango
ii  kms-core                              6.12.0-0kurento1.16.04                          amd64        Kurento Core module
ii  kms-core-dbg                          6.12.0-0kurento1.16.04                          amd64        Kurento Core module
ii  kms-elements                          6.12.0-0kurento1.16.04                          amd64        Kurento Elements module
ii  kms-elements-dbg                      6.12.0-0kurento1.16.04                          amd64        Kurento Elements module
ii  kms-filters                           6.12.0-0kurento1.16.04                          amd64        Kurento Filters module
ii  kms-filters-dbg                       6.12.0-0kurento1.16.04                          amd64        Kurento Filters module
ii  kms-jsonrpc                           6.12.0-0kurento1.16.04                          amd64        Kurento JSON-RPC library
ii  kms-jsonrpc-dbg                       6.12.0-0kurento1.16.04                          amd64        Kurento JSON-RPC library
ii  kmsjsoncpp                            1.6.3-1kurento1.16.04                           amd64        Kurento jsoncpp library
ii  kmsjsoncpp-dbg                        1.6.3-1kurento1.16.04                           amd64        Kurento jsoncpp library
ii  kurento-dbg                           6.12.0-0kurento1.16.04                          amd64        Meta-package that installs debug symbols
ii  kurento-media-server                  6.12.0-0kurento1.16.04                          amd64        Kurento Media Server
ii  kurento-media-server-dbg              6.12.0-0kurento1.16.04                          amd64        Kurento Media Server
ii  libgstreamer-plugins-bad1.5-0:amd64   1.8.1-1kurento4.16.04                           amd64        GStreamer development files for libraries from the "bad" set
ii  libgstreamer-plugins-base1.5-0:amd64  1.8.1-1kurento1.16.04                           amd64        GStreamer libraries from the "base" set
ii  libgstreamer1.5-0:amd64               1.8.1-1kurento2.16.04                           amd64        Core GStreamer libraries and elements
ii  libgstreamer1.5-0-dbg:amd64           1.8.1-1kurento2.16.04                           amd64        Core GStreamer libraries and elements
ii  libnice10:amd64                       0.1.15-1kurento3.16.04                          amd64        ICE library (shared library)
ii  libnice10-dbgsym:amd64                0.1.15-1kurento3.16.04                          amd64        debug symbols for package libnice10
ii  libsrtp0:amd64                        1.6.0-0kurento1.16.04                           amd64        Secure RTP (SRTP) and UST Reference Implementations - shared library
ii  libusrsctp                            0.9.2-1kurento1.16.04                           amd64        sctp-refimpl library
ii  openh264                              1.4.0-1kurento1.16.04                           amd64        OpenH264 library
ii  openh264-gst-plugins-bad-1.5:amd64    1.8.1-1kurento4.16.04                           amd64        GStreamer plugins from openh264
ii  openwebrtc-gst-plugins                0.10.0-1kurento1.16.04                          amd64        OpenWebRTC specific GStreamer plugins
ii  openwebrtc-gst-plugins-dbg            0.10.0-1kurento1.16.04                          amd64        OpenWebRTC specific GStreamer plugins

```

**System description:**

Please describe your setup (where is KMS located, where are the clients,
whether you are using STUN/TURN, etc.)
the default setup mentioned in https://openvidu.io/docs/deployment/deploying-ubuntu/

**What steps will reproduce the problem?**

Run the above setup for a while and create few rooms are participants,
 Don't have specific information on reproducing this issue

**What happens instead?**
segmentation fault and kms crash

**Does it happen with one of the tutorials?**
yes

**Comment by amardeep-s** [comment]
We are getting Segment Fault as well on Ubuntu 18.04 with KMS 6.13. Its crashing multiple times in a day.
Error in KMS log.

`Segmentation fault (thread 139911982216960, pid 852)
Stack trace:
[??]
??:0
0xCB96 at /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstdtls.so
[??]
??:0
0xE97E at /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstdtls.so
[??]
??:0
0x6134 at /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstdtls.so
[??]
??:0
0x63CA at /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstdtls.so
[gst_flow_get_name]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x6E527
[gst_pad_push]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x76553
[??]
??:0
0x9432 at /usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstdtls.so
[gst_flow_get_name]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x6E527
[gst_pad_push]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x76553
[gst_proxy_pad_chain_default]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x5F56B
[gst_flow_get_name]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x6E527
[gst_pad_push]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0x76553
[gst_base_src_wait_playing]
/usr/lib/x86_64-linux-gnu/libgstbase-1.5.so.0:0x30DD5
[gst_tag_setter_get_tag_merge_mode]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0xA0A00
[g_thread_pool_new]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x74B60
[g_test_get_filename]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x74195
`

**Comment by j1elo** [comment]
Hi @amardeep-s , please [install debug symbols](https://doc-kurento.readthedocs.io/en/latest/dev/dev_guide.html#install-debugging-symbols) so the stack traces include source file and line numbers, to see if your issue is the same as the one reported by @syamgk

**Comment by j1elo** [comment]
Hi @syamgk please [upgrade](https://doc-kurento.readthedocs.io/en/latest/user/installation.html#local-upgrade) to Kurento 6.13.0, to see if the problem persists.

Also, please enable [debug logging](https://doc-kurento.readthedocs.io/en/latest/features/logging.html) for this component:

```
dtlsdec:5,dtlsconnection:5
```

**Comment by amardeep-s** [comment]
@j1elo - I have tried installing debug symbols on 18.04 following instructions using dev guide, however there is a problem with **depends** in **kurento-dbg**

**Error** 

sudo apt-get update && sudo apt-get install --no-install-recommends --yes     kurento-dbg
Hit:1 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:2 http://archive.ubuntu.com/ubuntu bionic-updates InRelease                                                                                                                                      
Hit:3 http://archive.ubuntu.com/ubuntu bionic-backports InRelease                                                                                                                                    
Ign:4 http://ddebs.ubuntu.com bionic InRelease                                                                                                                                                       
Ign:5 http://ddebs.ubuntu.com bionic-updates InRelease                                                                                                                                               
Hit:6 http://ddebs.ubuntu.com bionic Release                                                                                                                                                                  
Hit:7 http://ddebs.ubuntu.com bionic-updates Release                                                                                                                                                          
Hit:8 https://deb.nodesource.com/node_10.x bionic InRelease                                                                                                                                          
Hit:9 http://ubuntu.openvidu.io/6.13.0 bionic InRelease                                                                                                                                              
Get:10 https://download.docker.com/linux/ubuntu bionic InRelease [64.4 kB]                                                                           
Hit:11 http://security.ubuntu.com/ubuntu bionic-security InRelease                                                             
Fetched 64.4 kB in 1s (72.9 kB/s)                  
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 kurento-dbg : Depends: libglib2.0-0-dbg but it is not installable
E: Unable to correct problems, you have held broken packages.


- kurento-dbg is looking for **libglib2.0-0-dbg** on bionic 18.04 but the package listed from ddebs is **libglib2.0-0-dbgsym**

> apt-cache policy libglib2.0-0-dbgsym
> libglib2.0-0-dbgsym:
>   Installed: (none)
>   Candidate: 2.56.4-0ubuntu0.18.04.5
>   Version table:
>      2.56.4-0ubuntu0.18.04.5 500
>         500 http://ddebs.ubuntu.com bionic-proposed/main amd64 Packages
>      2.56.4-0ubuntu0.18.04.4 500
>         500 http://ddebs.ubuntu.com bionic-updates/main amd64 Packages
>      2.56.1-2ubuntu1 500
>         500 http://ddebs.ubuntu.com bionic/main amd64 Packages
> 

- Another issue is that **libglib2.0-0-dbgsym** depends on **libglib2.0-0** which is coming from bionic-updates with version 2.56.4-0ubuntu0.18.04.4 see below.

Using ddebs it brings version **2.56.4-0ubuntu0.18.04.5** as you can see in above. 

**_To workaround this I had to disabled bionic-proposed first then install libglib2.0-0-dbgsym._**


> libglib2.0-0:
>   Installed: 2.56.4-0ubuntu0.18.04.4
>   Candidate: 2.56.4-0ubuntu0.18.04.4
>   Version table:
>  *** 2.56.4-0ubuntu0.18.04.4 500
>         500 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages
>         500 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages
>         100 /var/lib/dpkg/status
>      2.56.1-2ubuntu1 500
>         500 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages

Can you suggest what you want me to do ?

**Comment by j1elo** [comment]
> */To workaround this I had to disabled bionic-proposed first then
> install libglib2.0-0-dbgsym./*
>
This is the correct solution. I have done some tests with both Ubuntu
16.04 and 18.04, and the conclusion is the same: we don't want the
"-proposed" repository.

I've now updated the documentation to reflect this.

Also please note that the package "kurento-dbg" works fine for Ubuntu
16.04, but in 18.04 it fails. We noticed this yesterday, so I've fixed
it already but the fix hasn't been published yet.

Meanwhile, install the debug packages manually; this is the whole list:

libc6-dbg
libglib2.0-0-dbgsym | libglib2.0-0-dbg
kmsjsoncpp-dbg
libnice10-dbgsym | libnice-dbg
openwebrtc-gst-plugins-dbg
libgstreamer1.5-0-dbg
gstreamer1.5-libav-dbg
gstreamer1.5-nice-dbgsym
gstreamer1.5-plugins-bad-dbg
gstreamer1.5-plugins-base-dbg
gstreamer1.5-plugins-good-dbg
gstreamer1.5-plugins-ugly-dbg
kms-jsonrpc-dbg
kms-core-dbg
kms-elements-dbg
kms-filters-dbg
kurento-media-server-dbg

**Comment by amardeep-s** [comment]
@j1elo  - Thanks a lot. I have now installed all packages manually and will post back results on next crash.

**Comment by j1elo** [comment]
Anyway now that I notice, this one might be a duplicate of https://github.com/Kurento/bugtracker/issues/409

**Comment by amardeep-s** [comment]
@j1elo - We experienced crash yesterday. Please find the logs below.

**Error Logs**

```
2020-01-27T10:03:43+00:00 -- New execution
Segmentation fault (thread 139775927240448, pid 20821)
Stack trace:
[export_srtp_keys]
./ext/dtls/gstdtlsconnection.c:631
[gst_dtls_connection_process]
./ext/dtls/gstdtlsconnection.c:512
[process_buffer]
./ext/dtls/gstdtlsdec.c:498
[sink_chain]
./ext/dtls/gstdtlsdec.c:579
[gst_pad_chain_data_unchecked]
./gst/gstpad.c:4185
[gst_pad_push]
./gst/gstpad.c:4554
[sink_chain]
./ext/dtls/gstdtlssrtpdemux.c:128
[gst_pad_chain_data_unchecked]
./gst/gstpad.c:4185
[gst_pad_push]
./gst/gstpad.c:4554
[gst_proxy_pad_chain_default]
./gst/gstghostpad.c:127
[gst_pad_chain_data_unchecked]
./gst/gstpad.c:4185
[gst_pad_push]
./gst/gstpad.c:4554
[gst_base_src_loop]
./libs/gst/base/gstbasesrc.c:2851
[gst_task_func]
./gst/gsttask.c:346
[g_thread_pool_thread_proxy]
glib/gthreadpool.c:308
[g_thread_proxy]
glib/gthread.c:784

2020-01-27T10:04:28+00:00 -- New execution
(kurento-media-server:21344): GStreamer-CRITICAL **: 16:56:00.844: Element rtpbin259 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 16:56:00.846: Element rtpbin259 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:05:37.837: Element rtpbin267 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:05:37.838: Element rtpbin267 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:08:39.446: Element rtpbin273 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:08:39.448: Element rtpbin273 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:34:36.743: Element rtpbin281 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:34:36.747: Element rtpbin281 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:35:06.059: Element rtpbin284 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:35:06.061: Element rtpbin284 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:36:10.844: Element rtpbin286 already has a pad named send_rtp_sink_0, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
(kurento-media-server:21344): GStreamer-CRITICAL **: 17:36:10.845: Element rtpbin286 already has a pad named send_rtp_sink_1, the behaviour of  gst_element_get_request_pad() for existing pads is undefined!
```

**Comment by j1elo** [comment]
Hi @amardeep-s, can you run Kurento with GDB and get a backtrace information? That will help with the `GStreamer-CRITICAL` issue. Instructions here: [GStreamer-CRITICAL messages in the log](https://doc-kurento.readthedocs.io/en/latest/user/troubleshooting.html#gstreamer-critical-messages-in-the-log)

**Comment by j1elo** [comment]
Also please note that **kurento-utils-js** **6.12.0** and **6.13.0** have a bug that sends multiple videos and audios to the same WebRtcEndpoint (but only 1 video and 1 audio is supported by Kurento).

Please **if you use kurento-utils-js, make sure you use the latest 6.13.1** that solves the issue.

**Comment by amardeep-s** [comment]
@j1elo  - Sure I'll do the GDB and post back the results. Though looking into logs its coming very rarely.

We don't have kurento-utils-js

**Comment by j1elo** [comment]
The error

> GStreamer-CRITICAL **: Element rtpbin1 already has a pad named recv_rtp_sink_1

that could be seen in Kurento logs was (most of the time) due to mistakenly adding more than 1 video and/or 1 audio media stream track to the RTCPeerConection object in the Javascript side of the application. This would try to negotiate multiple videos or audios, and cause critical warnings inside the GStreamer code in KMS.

Of course if KMS is not able to handle multiple media tracks, it should reject the medias in the SDP negotiation, not crash or cause internal `GStreamer-CRITICAL` warnings. That bug was solved with commit https://github.com/Kurento/kms-core/commit/6a47630cba2aba4c74527f9757b705c342da5106, available since Kurento 6.13.1. Upgrade instructions can be found here: [Local Upgrade](https://doc-kurento.readthedocs.io/en/latest/user/installation.html#local-upgrade)

Please reopen is the crash is still happening with latest version of KMS.
